cilium/CFP-25694-move-cilium-cli-to-cilium-repo.md

```diff
@@ -0,0 +1,70 @@
+# CFP-25694: Move cilium/cilium-cli code into cilium/cilium repository
+
+**Code Name: Mega Ketchup**
+
+**SIG: [sig-cilium-cli]**
+
+**Begin Design Discussion:** 2023-05-25
+
+**Cilium Release:** 1.15
+
+**Authors:** Michi Mutsuzaki `<michi@isovalent.com>`
+
+## Summary
+
+Move [cilium/cilium-cli] code into [cilium/cilium] repository.
+
+## Motivation
+
+The main motivation for moving [cilium/cilium-cli] code into [cilium/cilium]
+repository is to minimize the feature development overhead for:
+
+- adding new tests to `cilium connectivity test`.
+- adding new tasks to `cilium sysdump`.
+
+## Goals
+
+* Move [cilium/cilium-cli] code into [cilium/cilium] main branch under `cilium-cli/` directory.
+* Establish a backporting process for changes inside `cilium-cli/` directory.
+* Don't disrupt the cilium-cli development and release processes.
+
+## Non-Goals
+
+* Rename the name of the `cilium-cli` binary from `cilium` to something else.
+* Release `cilium-cli` as a part of Cilium releases.
+
+## Proposal
+
+### Overview
+
+### Releasing cilium-cli
+
+We do not need to change the release process / cadence to accomplish the goals
+defined in this CFP. Continue releasing `cilium-cli` from the main branch at its
+own cadence using its own versioning that is independent of [cilium/cilium]
+versioning as we have been doing in [cilium/cilium-cli] repository.
+
+We need to come up with a git tag convention for cilium-cli releases to distinguish
+tags for Cilium releases from tags for cilium-cli releases. My proposal to use
+`cilium-cli/vX.Y.Z` for cilium-cli release tags.
+
+### Backporting to stable branches
+
+Starting in `v1.15`, we will need to backport bug fixes related to
+`cilium connectivity test` command from `main` branch to affected stable
+branches since these stable branches use their own copy of `cilium-cli/` to run
+`cilium connectivity test`. Apply the same [backporting process] as the rest of
+the Cilium codebase.
+
+## Impacts / Key Questions
+
+### Impact: Additional overhead for cilium-cli development
+
+### Key Question: Do we lose cilium-cli git history when we move cilium-cli code into cilium/cilium repository?
+
```

Comment:
Missing key question: How do we run CI for CLI? More specifically, how do we ensure that changes to Cilium CLI remain compatible with older released versions of Cilium?

At the moment, `cilium/cilium` CI only tests against `main`, we would have to have a CI workflow that tests the Cilium CLI main at least against the last Cilium CNI stable release, otherwise we'd loose CI coverage during the merge.

Reply:
added an answer 👍